### DISCUSSION TITLE
How to set a Constraint ?

### DISCUSSION THREAD

**Discussion by Franckapik** [pmndrs/use-cannon#90] [General] [answered]
I am sorry but i don't understand how to set a constraint, for example Hinge.

I don't really understand typescript yet, and wrote the code below but i am stuck in using the hook .

https://gist.github.com/Franckapik/0540c44ca05e07ae5573178ff5eb6793

Is that correct syntax ? How is it possible to add constraint to the world next ?  have you got example ? 

Thank you really much !!

F.

**Comment by Franckapik** [comment]
Ok i've got it !
I just have to write it like this : 
usePointToPointConstraint(cube, wheelShapeLF, { pivotA: [0, 0, 0], pivotB: [0, 2, 0] })
I just cloned the repo and see others examples like MondayMorning.

Just to understand, the examples are not linked in the docs because they are in development ? 
I hope i could help by trying to make a rigid car using constraints and forceImpulse according to cannon.js example.

**Comment by Franckapik** [comment]
And i just realized that "spring" could mean another thing than the season according to this usefull image :
![image](https://user-images.githubusercontent.com/21478769/85202474-f8965980-b306-11ea-9a57-5b8ea71cef7a.png)

:)

**Comment by codynova** [comment, marked as answer]
Hey @Franckapik, it looks like you got it figured out, but you had the `useHingeConstraint` function signature wrong. It should be:

```javascript
useHingeConstraint(bodyA, bodyB, options, deps)
```

These are described in the readme under the [API section](https://github.com/react-spring/use-cannon#api). It would probably be a good idea to mention the examples directory in the readme too.

```typescript
function useHingeConstraint(
  bodyA: React.MutableRefObject<THREE.Object3D>,
  bodyB: React.MutableRefObject<THREE.Object3D>,
  optns: HingeConstraintOpts,
  deps: any[] = [],
): ConstraintApi
```

**Comment by Franckapik** [comment]
Thank you @codynova for your help !
So now, i have created a car and try to move it with arrow key and applyImpulse.
My code is not really clean :/ but is there something wrong in the way i am using use-cannon ? 

Thank you again !

https://codesandbox.io/s/github/Franckapik/NandY/tree/master/views?file=/src/3d/jsx/Essai.js

I am applying impulse on rear wheel, and try to rotate front wheel to turn the car but it seems that i have an issue on this...
